src/sentry/tagstore/query.py

```diff
@@ -0,0 +1,40 @@
+from __future__ import absolute_import, print_function
+
+from django.db.models import sql
+from django.db.models.query import QuerySet
+from sentry.db.models import BaseManager
+
+
+class NoTransactionUpdateQuerySet(QuerySet):
+    def update(self, **kwargs):
+        """
+        Updates all elements in the current QuerySet, setting all the given
+        fields to the appropriate values.
+        """
+        # HACK(mattrobenolt): This is copy/pasted directly from
+        # https://github.com/django/django/blob/stable/1.6.x/django/db/models/query.py#L482-L496
+        # with 1 important change. It removes the `transaction.commit_on_success_unless_managed`
+        # block. The effect of this is we now can perform a simple `UPDATE` query without
+        # incurring the overhead of 4 statements and an explicit transaction. This is a safe
```

Comment:
Each UPDATE looked like:

```
LOG:  statement: SET default_transaction_isolation TO DEFAULT
LOG:  statement: BEGIN ISOLATION LEVEL READ COMMITTED
LOG:  statement: UPDATE "tagstore_tagvalue" SET "times_seen" = 1 WHERE ("tagstore_tagvalue"."project_id" = 1  AND "tagstore_tagvalue"."id" = 1 )
LOG:  statement: COMMIT
```

This means there are 4 round trip statements, 3 of which are extraneous. These extra statements add both to time of the request, as well as extra CPU on the databases and pgbouncer, etc. Considering we do _tons_ of UPDATES this should have a more than nominal impact.